src/java.base/share/classes/sun/nio/cs/StreamDecoder.java

```diff
@@ -72,11 +72,8 @@ public static StreamDecoder forInputStreamReader(InputStream in,
         throws UnsupportedEncodingException
     {
         String csn = charsetName;
-        if (csn == null) {
-            csn = Charset.defaultCharset().name();
-        }
         try {
-            return new StreamDecoder(in, lock, Charset.forName(csn));
+            return new StreamDecoder(in, lock, csn == null ? Charset.defaultCharset() : Charset.forName(csn));
```

Comment:
Can you check the callers, it may be that the mapping of null to the default CS can be removed.

Reply:
The callers throw NPE in case of null csName, but both are Encoder and Decoder are public. Will it be safe to remove explicit null check?

Reply:
sun.nio.cs is a JDK internal package. The only usages of StreamDecoder.forInputStreamReader and StreamEncoder.forOutputStreamWriter. You should be able to quickly check the usages.

Reply:
DOne

Reply:
Can you confirm that you've run jdk_tier2?

Reply:
Still running, NIO tests are ok
```
==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
   jtreg:test/jdk:jdk_nio                              475   475     0     0   
==============================
TEST SUCCESS
```

Reply:
The jdk_io test group will probably exercise this code the most, running test/jdk:tier2 is best for the case as it will run the jdk_io, jdk_nio and other tests that directly/indirectly make use of it.

Reply:
As of jdk:tier2 there are 4 failed tests, I suspect this is due to networking issues as I run them on virtual env
```
==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
>> jtreg:test/jdk:tier2                               4046  4032     4    10 <<
==============================
TEST FAILURE
```
```
JT Harness : Tests that failed
com/sun/jndi/ldap/LdapPoolTimeoutTest.java: Multi-threaded client timeout tests for ldap pool
java/net/InetAddress/IsReachableViaLoopbackTest.java: ensure we can't ping external hosts via loopback if
java/net/InetAddress/getOriginalHostName.java: test functionality of getOriginalHostName(InetAddress)
java/net/MulticastSocket/Promiscuous.java: Test for interference when two sockets are bound to the same port but joined to different multicast groups
```